spacy/pipeline/edit_tree_lemmatizer.py

```diff
@@ -150,7 +149,7 @@ def predict(self, docs: Iterable[Doc]) -> List[Ints2d]:
             # Handle cases where there are no tokens in any docs.
             n_labels = len(self.cfg["labels"])
             guesses: List[Ints2d] = [
-                self.model.ops.alloc((0, n_labels), dtype="i") for doc in docs
+                self.model.ops.alloc2i(0, n_labels, dtype="i") for _ in docs
```

Comment:
If these two lines aren't equivalent I feel like the error might still be in `thinc`?

```
self.model.ops.alloc((0, n_labels), dtype="i")
self.model.ops.alloc2i(0, n_labels)
```

Reply:
The issue was the before 8.1.3, `ops.alloc` wasn't properly typed. Now it is properly typed and the error that is being thrown is actually a reasonable one by `mypy`. If you want to ensure you get a specific type back (like here, expecting a `List[2d]`) you can't expect `op.alloc` to work that out based on the tuple format and the `dtype` - that's why we have the more specific methods, right?

Reply:
So these lines are equivalent in functionality, but not in typing behaviour.

Reply:
For the future, I think we should try to make them equivalent in typing behavior, too, or this will be confusing for users.